src/lib/weather_vane/WeatherVane.cpp

```diff
@@ -41,24 +41,45 @@
 #include <mathlib/mathlib.h>
 
 
-WeatherVane::WeatherVane() :
-	ModuleParams(nullptr)
+WeatherVane::WeatherVane(ModuleParams *parent) :
+	ModuleParams(parent)
 { }
 
-void WeatherVane::update(const matrix::Vector3f &dcm_z_sp_prev, float yaw)
+void WeatherVane::update()
 {
-	_dcm_z_sp_prev = dcm_z_sp_prev;
-	_yaw = yaw;
+	vehicle_status_s vehicle_status;
+
+	if (_vehicle_status_sub.update(&vehicle_status)) {
+		_is_vtol = vehicle_status.is_vtol;
+	}
+
+	vehicle_control_mode_s vehicle_control_mode;
+
+	if (_vehicle_control_mode_sub.update(&vehicle_control_mode)) {
+		_flag_control_manual_enabled = vehicle_control_mode.flag_control_manual_enabled;
+		_flag_control_position_enabled = vehicle_control_mode.flag_control_position_enabled;
+	}
+
+	// Weathervane is only enabled for VTOLs if it's enabled by parameter
+	// in manual we use weathervane just if position is controlled as well
+	// in mission we use weathervane except for when navigator disables it
+	_is_active = _is_vtol && _param_wv_en.get()
```

Comment:
What do you think about removing the if(vtol) check here, and thus enable WV to be used for MC as well? It could be useful for asymmetric multicopters that gain stability when aligning their nose into the wind (e.g. if they have a larger drag coefficient in y compared to x body direction).

Reply:
Great! 

I guess the simplest check would be to simulate it in the [__windy gazebo environment with a high wind speed](https://docs.px4.io/v1.12/en/simulation/gazebo.html#change-wind-speed), let the drone takeoff with the nose maybe 90° from the wind direction and then see if it aligns with the wind once the takeoff altitude is reached. 

I guess the idea is that if WV is enabled, all the `position_setpoint.yaw `are overwritten by it, unless the `position_setpoint.disable_weather_vane` is true (e.g. required for aligning a VTOL to it's transition direction)? I don't see right now where this workflow wouldn't work for a MC as well, there `position_setpoint.disable_weather_van`e is likely just always false as I don't see when a MC that has WV enabled should suddenly care about it's yaw. It's either you don't care about yaw or you do and you don't enable WV (as a system setting, not a per-waypoint thingy). Thoughts?